code/modules/multiz/movement.dm

```diff
@@ -208,29 +234,112 @@
 	if(locate(/obj/structure/stairs) in landing)
 		return 1
 
+	var/area/area1 = get_area(landing)
+	if(!area1.has_gravity())
+		return 1
+
+	if(istype(src,/mob/living) || isobj(src))
+		if(istype(landing, /turf/simulated/open))
+			z_levels_fallen++ // Don't damage them but keep track of this.
+			return 1
+
+		if(istype(landing, /turf/space))
+			z_levels_fallen = 0 // turns out they didn't hit anything solid. Lucky them.
+			return 1
+
+		if(!z_levels_fallen) // Checks if they only fell down one floor.
+			z_levels_fallen = 1
+
+		var/weight = 1
+		if(ismob(src))
+			var/mob/M = src
+			weight = M.mob_size
+		if(isobj(src))
+			var/obj/O = src
+			weight = O.w_class
+
+
+		var/fallforce = throw_range
+		if(isobj(src))
+			var/obj/O = src
+			fallforce = O.throwforce
+
+		if(weight >= 3 && fallforce <= 5)
+			fallforce = throw_range
+
+		var/speed = z_levels_fallen*throw_speed
+		var/mass = (weight/THROWNOBJ_KNOCKBACK_DIVISOR)+density+opacity
+		var/momentum = speed*mass
+		var/damage = fallforce*(speed/THROWFORCE_SPEED_DIVISOR)*momentum
+
+		var/miss_chance = max(15*(z_levels_fallen-1), 0)
+
+		if(!prob(miss_chance))
+			for(var/mob/living/L in landing)
```

Comment:
I have decided that only hitting one mob was arbitrary, and if that there are multiple mobs on a single tile (which is relatively) uncommon, they should all be hit.